engine/api/application_deployment.go

```diff
@@ -17,12 +17,18 @@ import (
 func (api *API) getApplicationDeploymentStrategiesConfigHandler() service.Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
 		vars := mux.Vars(r)
-		key := vars[permProjectKey]
+		projectKey := vars[permProjectKey]
 		appName := vars["applicationName"]
 
-		app, err := application.LoadByName(api.mustDB(), key, appName, application.LoadOptions.WithDeploymentStrategies)
+		proj, err := project.Load(api.mustDB(), projectKey)
```

Comment:
I think it's a bit sad to load the project to get its ID, whereas the projectKey is an identifier.
we should consider `application.LoadByProjectKeyAndName`